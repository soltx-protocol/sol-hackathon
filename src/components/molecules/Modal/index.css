.modal {
	display: flex;
	justify-content: center;
	width: 100vw;
	height: 100%;
	position: fixed;
	top: 0;
	left: 0;
	overflow: auto;

	/* overlay the top header(100) */
	z-index: 200;

	&.inline {
		position: absolute;
	}
}
