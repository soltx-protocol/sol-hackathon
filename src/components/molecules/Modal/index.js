/* eslint-disable react/jsx-props-no-spreading */

import React, { useLayoutEffect, useRef } from 'react';
import ReactDOM from 'react-dom';

import classnames from 'classnames';

import styles from './index.css';

const ModalComponent = ({ className, children, active, inline, onDeactive }) => {
	if (!active) {
		return null;
	}

	return (
		<div
			className={classnames(styles.modal, { [styles.inline]: inline }, className)}
			onClick={e => {
				if (e.target === e.currentTarget) {
					onDeactive(e);
				}
			}}
			role="presentation"
		>
			{children}
		</div>
	);
};

export const InlineModal = ({ children, ...other }) => (
	<ModalComponent inline {...other}>
		{children}
	</ModalComponent>
);

const Modal = ({ children, ...other }) => {
	const refDiv = useRef(document.createElement('div'));

	useLayoutEffect(() => {
		let modalRoot = document.getElementById('modal-root');

		if (modalRoot === null) {
			modalRoot = document.createElement('div');
			modalRoot.setAttribute('id', 'modal-root');
			document.body.appendChild(modalRoot);
		}

		modalRoot.appendChild(refDiv.current);

		return () => {
			modalRoot.removeChild(refDiv.current);
		};
	}, []);

	return ReactDOM.createPortal(
		<ModalComponent {...other}>{children}</ModalComponent>,
		refDiv.current,
	);
};

export default Modal;
