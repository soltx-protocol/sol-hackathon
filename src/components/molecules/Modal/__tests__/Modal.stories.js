import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs/react';

import Modal, { InlineModal } from 'components/molecules/Modal';

const stories = storiesOf('COMMON|molecules/Modal', module);

stories.add('__interactive', () => (
	<Modal active={boolean('active', true)}>
		<div style={{ backgroundColor: 'black', width: '100px', height: '50px', color: 'white' }}>
			測試
		</div>
	</Modal>
));

stories.add('with inline', () => (
	<InlineModal active={boolean('active', true)}>
		<div style={{ backgroundColor: 'black', width: '100px', height: '50px', color: 'white' }}>
			測試
		</div>
	</InlineModal>
));
