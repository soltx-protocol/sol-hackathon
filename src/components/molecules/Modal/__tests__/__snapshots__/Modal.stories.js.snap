// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `<DocumentFragment />`;

exports[`with inline 1`] = `
<DocumentFragment>
  <div
    class="modal inline"
    role="presentation"
  >
    <div
      style="background-color: black; width: 100px; height: 50px; color: white;"
    >
      測試
    </div>
  </div>
</DocumentFragment>
`;
